<a href="https://colab.research.google.com/github/OlegV12/Stats/blob/Lesson_1/mathStats.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [ ]:
from math import factorial
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [ ]:
# all clubs combination
all_clubs = combinations(13, 4)
all_clubs

715

In [ ]:
# all 4 cards combinations
total = combinations(52, 4)
total

270725

In [ ]:
# probability
P = all_clubs / total
P

0.0026410564225690276

а) Ответ: 0.0026410564225690276 или 715/270725

In [35]:
# inverted probability of no aces among 4 cards 
1 - (combinations(48, 4) / total)

0.2812632745405854

б) Ответ: 0.2812632745405854

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [40]:
lock_comb = combinations(10, 3)
lock_comb

120

Ответ: 1/120

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [38]:
total_tools_combinations = combinations(15, 3)
painted_tools_combinations = combinations(9, 3)
total_tools_combinations, painted_tools_combinations

(455, 84)

In [39]:
result = painted_tools_combinations / total_tools_combinations
result


0.18461538461538463

Ответ: 84/455 или 0.18461538461538463

В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [ ]:
2/100 * 2/100

0.0004

Ответ: 0.0004